src/hotspot/share/compiler/methodMatcher.cpp

```diff
@@ -269,6 +269,19 @@ void MethodMatcher::parse_method_pattern(char*& line, const char*& error_msg, Me
     c_match = check_mode(class_name, error_msg);
     m_match = check_mode(method_name, error_msg);
 
+    // Over-consumption
+    // method_name points to an option type or option name because the method name is not specified by users.
+    // In very rare case, the method name happens to be same as option type/name, so look ahead to make sure
+    // it doesn't show up again.
+    if ((OptionType::Unknown != CompilerOracle::parse_option_type(method_name) ||
+        CompileCommand::Unknown != CompilerOracle::parse_option_name(method_name)) &&
+        *(line + bytes_read) != '\0' &&
+        strstr(line + bytes_read, method_name) == NULL) {
+      error_msg = "Did not specify any method name";
+      m_match = MethodMatcher::Unknown;
+      method_name[0] = '\0';
```

Comment:
Why don't you just return here as in the error cases below? If you would do something like `-XX:CompileCommand=option,"*<",ccstrlist,ControlIntrinsic,+_id`, the error message would be overwritten below.

Reply:
`method_name[0] = '\0` can also be removed. Otherwise, it looks good to me!